pkg/drivers/virtiofs.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2025 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package drivers
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// VirtiofsShare is a directory on the host shared with the guest using
+// virtiofs.
+type VirtiofsShare struct {
+	SharedDir string
+	MountTag  string
+}
+
+// ParseVirtiofsShare parses --virtiofs-share argument and returns a validated VirtiofsShare.
+// - SharedDir is absolute path to existing directroy to share with the guest.
+// - MountTag is non empty string without leading or trailing whitespace.
+func ParseVirtiofsShare(s string) (*VirtiofsShare, error) {
```

Comment:
Should validate all virtuous shares, since we can have duplicate or conflicting shares. This will also simplify the caller code.

```go
ValidateVirtiofsShare([]string]) ([]VirtiofsShare, error)
```